io/plugins/eu.esdihumboldt.hale.io.shp.test/src/eu/esdihumboldt/hale/io/shp/ShapefileInstanceWriterTest.groovy

```diff
@@ -184,11 +186,25 @@ class ShapefileInstanceWriterTest {
 			println "Temporary file is $tmpFile"
 			writeInstances(tmpFile.toFile(), schema, instances, configurator)
 			handler.accept(tmpFile.toFile())
+
+			testCpgFileAgainstRelatedSpapefile(tmpFile.toFile())
 		} finally {
 			tmpDir.deleteDir()
 		}
 	}
 
+	private static testCpgFileAgainstRelatedSpapefile(File shpFile) {
+		ShapefileDataStore store = new ShapefileDataStore(shpFile.toURL());
+		String cpgFilePath = shpFile.getAbsolutePath().replace(ShapefileConstants.SHP_EXTENSION, ShapefileConstants.CPG_EXTENSION)
+
+		def cpgFile = new File(cpgFilePath)
+		if (cpgFile.exists()) {
+			assertTrue(store.getCharset().toString().equals(cpgFile.text))
```

Comment:
As mentioned in the comment on the writer class, the `store.getCharset()` here will always just return the default value for the charset that is specified in the `ShapefileDataStore` class, and thus this check only works because anyway always the same charset is used for writing the files.